src/common/query/src/columnar_value.rs

```diff
@@ -40,16 +40,12 @@ impl ColumnarValue {
 
     /// Convert a columnar value into an VectorRef
     pub fn try_into_vector(self, num_rows: usize) -> Result<VectorRef> {
-        Ok(match self {
-            ColumnarValue::Vector(v) => v,
+        match self {
+            ColumnarValue::Vector(v) => Ok(v),
             ColumnarValue::Scalar(s) => {
-                let v = s
-                    .to_array_of_size(num_rows)
-                    .context(GeneralDataFusionSnafu)?;
-                let data_type = v.data_type().clone();
-                Helper::try_into_vector(v).context(IntoVectorSnafu { data_type })?
+                Helper::try_from_scalar_value(s, num_rows).context(FromScalarValueSnafu)
```

Comment:
We can replace the implementation of try_from_scalar_value() with `to_array_of_size() + try_into_vector()` as it should support more types.

Reply:
I don't think so. `try_from_scalar_value` use constant vector, which is better than `to_array_of_size() + try_into_vector()`.

Reply:
Looks like the unsupported list in these two methods are almost the same

https://github.com/GreptimeTeam/greptimedb/blob/main/src/datatypes/src/vectors/helper.rs#L237-L247

https://github.com/GreptimeTeam/greptimedb/blob/main/src/datatypes/src/vectors/helper.rs#L374-L386

I don't think we should change this behaviour.

Reply:
We have many calls to `downcast_ref` on vectors that don't check if it's constant.

Looks like it's better to revert this change.